include/fi_list.h

```diff
@@ -107,11 +107,10 @@ static inline void dlist_remove(struct dlist_entry *item)
     for ((item) = (head)->next, (tmp) = (item)->next; (item) != (head);	\
               (item) = (tmp), (tmp) = (item)->next)
 
-#define dlist_foreach_container_safe(head, container, member, tmp)			\
-	for (container = container_of((head)->next, typeof(*container), member),	\
-	     (tmp) = container->member.next; &(container->member) != (head);		\
-	     container = container_of((tmp),						\
-				      typeof(*container), member),			\
+#define dlist_foreach_container_safe(head, type, container, member, tmp)	\
+	for (container = container_of((head)->next, type, member),		\
+	     (tmp) = container->member.next; &(container->member) != (head);	\
+	     container = container_of((tmp), type, member),			\
```

Comment:
Why is this change needed?

Reply:
No Windows compilers don't support `typeof()` C operator. So, if we'd like to use this macro on Windows, we should remove this dependency - the same was done for `dlist_foreach_container`